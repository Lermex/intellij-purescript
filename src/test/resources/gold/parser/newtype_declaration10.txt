Purescript File
  PSASTWrapperElement(Program)
    PSASTWrapperElement(Module)
      PsiElement(module)('module')
      PsiElement(whitespace)(' ')
      PSASTWrapperElement(ModuleName)
        PSASTWrapperElement(Qualified)
          PsiElement(proper name)('A')
      PsiElement(whitespace)(' ')
      PsiElement(where)('where')
      PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(NewtypeDeclaration)
        PsiElement(newtype)('newtype')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeConstructor)
          PSASTWrapperElement(TypeConstructor)
            PsiElement(proper name)('Ta')
            PsiElement(whitespace)(' ')
        PsiElement(identifier)('a')
        PsiElement(whitespace)(' ')
        PsiElement(=)('=')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeConstructor)
          PSASTWrapperElement(TypeConstructor)
            PsiElement(proper name)('Ta')
            PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeAtom)
          PsiElement(()('(')
          PSASTWrapperElement(Type)
            PSASTWrapperElement(TypeAtom)
              PSASTWrapperElement(TypeConstructor)
                PSASTWrapperElement(Qualified)
                  PSASTWrapperElement(TypeConstructor)
                    PsiElement(proper name)('Eff')
                    PsiElement(whitespace)(' ')
            PSASTWrapperElement(TypeAtom)
              PsiElement(()('(')
              PSASTWrapperElement(Row)
                PSASTWrapperElement(LocalIdentifier)
                  PSASTWrapperElement(identifier)
                    PsiElement(identifier)('console')
                  PsiElement(whitespace)(' ')
                PsiElement(::)('::')
                PsiElement(whitespace)(' ')
                PSASTWrapperElement(Type)
                  PSASTWrapperElement(TypeAtom)
                    PSASTWrapperElement(TypeConstructor)
                      PSASTWrapperElement(Qualified)
                        PSASTWrapperElement(TypeConstructor)
                          PsiElement(proper name)('CONSOLE')
              PsiElement())(')')
              PsiElement(whitespace)(' ')
            PSASTWrapperElement(TypeAtom)
              PSASTWrapperElement(GenericIdentifier)
                PsiElement(identifier)('a')
          PsiElement())(')')