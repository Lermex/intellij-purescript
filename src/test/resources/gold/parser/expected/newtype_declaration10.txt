Purescript File
  PSASTWrapperElement(Program)
    PSASTWrapperElement(Module)
      PsiElement(module)('module')
      PsiElement(whitespace)(' ')
      PSASTWrapperElement(Qualified)
        PSASTWrapperElement(ModuleName)
          PsiElement(proper name)('A')
      PsiElement(whitespace)(' ')
      PsiElement(where)('where')
      PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(NewtypeDeclaration)
        PsiElement(newtype)('newtype')
        PsiElement(whitespace)(' ')
        PsiElement(proper name)('Ta')
        PsiElement(whitespace)(' ')
        PsiElement(identifier)('a')
        PsiElement(whitespace)(' ')
        PsiElement(=)('=')
        PsiElement(whitespace)(' ')
        PsiElement(proper name)('Ta')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeAtom)
          PsiElement(()('(')
          PSASTWrapperElement(Type)
            PSASTWrapperElement(TypeAtom)
              PSASTWrapperElement(TypeConstructor)
                PSASTWrapperElement(Qualified)
                  PsiElement(proper name)('Eff')
                  PsiElement(whitespace)(' ')
            PSASTWrapperElement(TypeAtom)
              PsiElement(()('(')
              PSASTWrapperElement(Row)
                PSASTWrapperElement(identifier)
                  PsiElement(identifier)('console')
                PsiElement(whitespace)(' ')
                PsiElement(::)('::')
                PsiElement(whitespace)(' ')
                PSASTWrapperElement(Type)
                  PSASTWrapperElement(TypeAtom)
                    PSASTWrapperElement(TypeConstructor)
                      PSASTWrapperElement(Qualified)
                        PsiElement(proper name)('CONSOLE')
              PsiElement())(')')
              PsiElement(whitespace)(' ')
            PSASTWrapperElement(TypeAtom)
              PSASTWrapperElement(TypeVar)
                PsiElement(identifier)('a')
          PsiElement())(')')